src/Compilers/Core/Portable/PEWriter/InheritedTypeParameter.cs

```diff
@@ -56,7 +56,7 @@ public bool MustHaveDefaultConstructor
 
         public TypeParameterVariance Variance
         {
-            get { return _parentParameter.Variance; }
+            get { return _inheritingType.IsInterface || _inheritingType.IsDelegate ? _parentParameter.Variance : TypeParameterVariance.NonVariant; }
```

Comment:
Not sure I understand this change. In what situation is the inheriting type not an interface or a delegate, but the parentParameter variance is variant, and that is incorrect?

Reply:
> Not sure I understand this change. In what situation is the inheriting type not an interface or a delegate, but the parentParameter variance is variant, and that is incorrect?

A class nested into an interface (for example, a display class) is inheriting type parameters of the containing interface. Those type parameters could be variant, but only interfaces and delegates are allowed to have variant type parameters.

---
In reply to: [347048798](https://github.com/dotnet/roslyn/pull/39839#discussion_r347048798) [](ancestors = 347048798)

Reply:
Got it, thanks.